From: Johannes Berg <johannes.berg@intel.com>
Date: Mon, 27 Jun 2022 12:49:03 +0200
Subject: [PATCH] wifi: cfg80211: remove chandef check in cfg80211_cac_event()
References: bsc#1012628
Patch-mainline: 5.19.2
Git-commit: d6f671c8a339d5b655acfacb8be6918c744fbabf

commit d6f671c8a339d5b655acfacb8be6918c744fbabf upstream.

The current check only worked for AP mode, but we can do
radar detection in mesh as well (for example). We could
try to check this using wdev_chandef(), but we also don't
really care since the chandef is passed in and we have no
need to use it anymore (since we added the argument in
commit d2859df5e7f0 ("cfg80211/mac80211: DFS setup chandef
for cac event")).

Change-Id: I856e4344d5e64ff4d2eead0b4c53b11f264be9b8
Fixes: 7b0a0e3c3a88 ("wifi: cfg80211: do some rework towards MLO link APIs")
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 net/wireless/mlme.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/net/wireless/mlme.c b/net/wireless/mlme.c
index fab2d6206cdd..b9204d0f1e55 100644
--- a/net/wireless/mlme.c
+++ b/net/wireless/mlme.c
@@ -955,9 +955,6 @@ void cfg80211_cac_event(struct net_device *netdev,
 	if (WARN_ON(!wdev->cac_started && event != NL80211_RADAR_CAC_STARTED))
 		return;
 
-	if (WARN_ON(!wdev->links[0].ap.chandef.chan))
-		return;
-
 	switch (event) {
 	case NL80211_RADAR_CAC_FINISHED:
 		timeout = wdev->cac_start_time +
-- 
2.37.2

